lib/DatabaseLayer.py

```diff
@@ -441,8 +441,10 @@ def getSearchResults(search):
     result = {"data": []}
     regSearch = re.compile(re.escape(search), re.IGNORECASE)
     links = {"n": "Link", "d": []}
-    for vLink in getInfo("via4").get("searchables", []):
-        links["d"].extend(sanitize(colVIA4.find({vLink: {"$in": [regSearch]}})))
+    via4 = getInfo("via4")
+    if via4:
+        for vLink in via4.get("searchables", []):
+            links["d"].extend(sanitize(colVIA4.find({vLink: {"$in": [regSearch]}})))
```

Comment:
By adding the check this way there is an additional query needed to the database (which btw doesn't actually check if the via4 collection is filled, it just checks if the via document is present in the info collection). IMO it would be better to wrap the for loop in a try; except, something like:

```suggestion
    try:
        for vLink in via4.get("searchables", []):
            links["d"].extend(sanitize(colVIA4.find({vLink: {"$in": [regSearch]}})))
    except AttributeError:
        return result
```

Reply:
@P-T-I  I don't understand why there would be an additional query.
The current master branch this query is performed on line 444
```
for vLink in getInfo("via4").get("searchables", []):
```

Reply:
On your line 444; via4 = getInfo("via4")

The getInfo (based in lib/DatabaseLayer.py) method makes the additional query to the database

Reply:
the getInfo("via4") needs to be performed either way. Otherwise there is no data to get the "searchables" from.

Reply:
You are absolutely right from the start; my bad 🤦‍♂️